sdk/state/open_test.go

```diff
@@ -222,17 +222,15 @@ func TestConfirmOpen_rejectsDifferentOpenAgreements(t *testing.T) {
 		// invalid ObservationPeriodTime
 		d := oa
 		d.ObservationPeriodTime = 0
-		_, authorized, err := channel.ConfirmOpen(OpenAgreement{Details: d})
-		require.False(t, authorized)
+		err := channel.validateOpen(OpenAgreement{Details: d})
```

Comment:
I missed this on my first pass: I think it is more important we test the exported interface of our SDK by calling `ConfirmOpen`. I think the result should be equivalent since the open is the first thing that takes place in the ConfirmOpen. What's your motivation for changing it?

Reply:
motivation was doing a unit test of `validateOpen` instead of `ConfirmOpen` which is doing more than just validating. One of the reasons I wanted to move these validation checks into their own methods was so we could improve the unit tests. Ie able to test just validation and not everything the `Confirm<>` methods are doing

wdyt?

Reply:
It's a trade off, and I don't think the pay off is worth the cost at this early point. It is easier to perform sweeping refactors and know that what we have still does the same thing it did before if we stick to testing, where possible, the exported surface area. I don't think it is worth optimizing that away yet. Because of that I'd focus on testing the exported interface rather than internals, and focus on behavior driven tests. I think the refactor is still good, but the tests should be able to stay the same.

Reply:
fair enough making tests more specific may be preemptive atm, changed back 👍 
https://github.com/stellar/experimental-payment-channels/pull/183/commits/1ccb79048b9b74d49ff7d071858b5633c76adbb7